ha/core/event_manager/event_manager.py

```diff
@@ -318,6 +318,19 @@ def get_events(self, component : str) -> list:
             list: List of events.
         """
 
+        key = f'{const.COMPONENT_KEY}/{component}'
+        value = []
+        Log.debug(f"Fetching subscribed events for {key}")
+
+        kv = self._confstore.get(key)
+        if kv:
+            for k, v in kv.items():
```

Comment:
In both function do we need for loop 

are they returning multiple value

Reply:
addressing case where keys with same prefix might get added 
ex: "message_type/sspl  , "message_type/sspl_test"  etc
OR the case where passed component string is incomplete ex:  function call looks like message_type("har")
       This should not return the list meant for "hare"

Reply:
_confstore.get can return multiple KVs ; where  "key" is the matching prefix. So for loop is needed